paddle/go/cmd/master/master.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"net/rpc"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/PaddlePaddle/Paddle/paddle/go/master"
+	"github.com/wangkuiyi/recordio"
+)
+
+const (
+	taskTimeoutDur = 20 * time.Minute
+	taskTimeoutMax = 3
+)
+
+func main() {
+	port := flag.Int("p", 0, "port of the master server")
+	dataset := flag.String("d", "", "dataset: comma separated path to RecordIO files")
+	faultTolerant := flag.Bool("fault-tolerance", false, "enable fault tolerance (requires etcd).")
+	flag.Parse()
```

Comment:
@helinwang 

> the best practices for building cloud-native applications

The most famous one is "12-factor" app: https://12factor.net 

For apps on kubernetes, there are generally 3 ways to pass arguments to program:

1. Via environment variables, kubernetes can also pass pod dynamic status info through environment variables using [downwardAPI](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/)
1. Via command-line arguments, specify arguments manually
1. Via configuration files through mounting a "config map" to pod